FunctionalComponentExample.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+import React, { useState, useEffect } from 'react';
+
+import {
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  Linking,
+  View,
+} from 'react-native';
+
+import QRCodeScanner from 'react-native-qrcode-scanner';
+
+
+const ScanQRCodeScreen = () => {
+  const [qrcodeGetData, setQRCodeGetData] = useState(null);
+  
+  useEffect(() => {
+  },[qrcodeGetData]);
```

Comment:
What is the purpose of an empty effect?

Reply:
Hi! From my understanding is to provide a realtime update on the screen when the value betewwn `[]` changes.

If the `useEffect ` Hook is not applied here, the change in `qrcodeGetData ` will be updated just after the next re-render.

Reply:
it's not actually true as long as React does that for you already. Once you have your state updated, the component will immediately re-render by himself

Reply:
HI! It just not worked for me. COuld you post an exemple? I mean it not work at the same time the `setQRCodeGetData` was executed

Reply:
what exactly didn't work? once you're reading a QR code, the `onSuccess` function has been called without `useEffect`?